src/main/java/mate/academy/spring/service/impl/CustomUserDetailService.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.spring.service.impl;
+
+import static org.springframework.security.core.userdetails.User.withUsername;
+
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailService implements UserDetailsService {
+    private UserService userService;
+
+    public CustomUserDetailService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username)
+            throws UsernameNotFoundException {
+        User user = userService.findByEmail(username).orElseThrow(() -> {
+            throw new UsernameNotFoundException("User not found");
+        });
+        UserBuilder builder = withUsername(username);
+        builder.password(user.getPassword());
+        builder.authorities(user.getRoles().stream()
+                .map(x -> x.getRoleName().name()).toArray(String[]::new));
```

Comment:
use roles instead of authorities and check the difference, and each stream method should be called on the new line